File: insert_sort/python/insertsort.py
def insertsort(data):
    for j in xrange(len(data) - 1):
        i = j
        res = data[j]
        while i > 0 and data[i-1] > res:
            data[i] = data[i-1]
            i -= 1

        data[i] = res

    return data


print insertsort([4, 1, 3, 2, 7, 9])

File: insert_sort/go/insertsort.go
package main

import (
	"fmt"
)

func main() {
	data := []int{4, 2, 5, 1, 7, 3, 5, 9}
	fmt.Println(data)
	fmt.Println(insertsort(data))
	fmt.Println(data)
}

func insertsort(data []int) []int {
	for pos, val := range data {
		i := pos
		for i > 0 && data[i-1] > val {
			data[i] = data[i-1]
			i--
		}

		data[i] = val
	}

	return data
}
